java/core/src/main/java/com/google/protobuf/GeneratedMessage.java

```diff
@@ -1396,7 +1396,7 @@ private void verifyExtensionContainingType(
       return setExtension((ExtensionLite<MessageType, Type>) extension, value);
     }
     /** Set the value of an extension. */
-    public final <Type> BuilderType setExtension(
```

Comment:
No, removing final is binary compatible:
https://docs.oracle.com/javase/specs/jls/se7/html/jls-13.html#jls-13.4.17